src/Tools/dotnet-monitor/DiagnosticPortValidateOptions.cs

```diff
@@ -12,8 +14,28 @@ namespace Microsoft.Diagnostics.Tools.Monitor
     internal class DiagnosticPortValidateOptions :
         IValidateOptions<DiagnosticPortOptions>
     {
+        private readonly CollectionRulesConfigurationProvider _provider;
+        private readonly ILogger<DiagnosticPortValidateOptions> _logger;
+
+        public DiagnosticPortValidateOptions(CollectionRulesConfigurationProvider provider, ILogger<DiagnosticPortValidateOptions> logger)
+        {
+            _provider = provider;
+            _logger = logger;
+        }
+
         public ValidateOptionsResult Validate(string name, DiagnosticPortOptions options)
         {
+            var collectionRuleNames = _provider.GetCollectionRuleNames();
+
+            if (collectionRuleNames != null && collectionRuleNames.Count > 0)
```

Comment:
These changes should not exist here. This class should only concern itself with validating the `DiagnosticPortOptions` instance. The changes in the `CollectionRuleService` should be sufficient to warn about the condition.

Reply:
I would tend to agree - I didn't really want to stick it here, but my understanding is that the changes to `CollectionRuleService` will only print when the configuration is changed (not on first load). If someone starts up `dotnet monitor` in `Connect` mode and never tries to change their config file during its execution, presumably they'll never get the warning, which feels like we aren't actually fixing the problem in a lot of user scenarios. I had previously looked at putting the warning in `ServerEndpointInfoSource.cs` when we do the check within `ExecuteAsync`, but felt like that file shouldn't need to have knowledge of the existence of collection rules.

Reply:
I see. The GH UI made me think the condition check and logging this was in the ctor. I'll add another comment for what I think should be changed over there.